implementation

berkebakkalogluosf opened the pull request.

berkebakkalogluosf committed:
implementation

berkebakkalogluosf committed:
major overhaul

romanbychkoosf commented on Opportunity Summarizer/force-app/main/default/classes/OpportunityMailSchedulerTest.cls:
```diff
@@ -0,0 +1,114 @@
+        
+        System.assertEquals(OpportunityMailScheduler.DAILY_CRON_EXP, ct.CronExpression);
+        System.assertEquals(0, ct.TimesTriggered);
+        System.assertEquals('2021-08-27 10:00:00', String.valueOf(ct.NextFireTime)); // Tested In: 2021-08-26 17:00:00
```
those asserts have to be after Test.stopTest();

romanbychkoosf commented on Opportunity Summarizer/force-app/main/default/classes/OpportunityMailSchedulerTest.cls:
```diff
@@ -0,0 +1,114 @@
+        CronTrigger ct = [SELECT Id, CronExpression, TimesTriggered, NextFireTime 
+                          FROM CronTrigger WHERE Id = :jobId];
+        
+        System.assertEquals(OpportunityMailScheduler.WEEKLY_CRON_EXP, ct.CronExpression);
```
those asserts have to be after Test.stopTest();

romanbychkoosf commented on Opportunity Summarizer/force-app/main/default/classes/OpportunitySummarizer.cls:
```diff
@@ -0,0 +1,147 @@
+        try {  
+            return Database.getQueryLocator('SELECT Id, AccountId, StageName, Amount ' +
+                                            'FROM Opportunity WHERE StageName IN :STAGES_TO_REPORT ' + 
+                                            'AND (CreatedDate = THIS_MONTH OR LastActivityDate = THIS_MONTH)');
```
it is better to use LastModifiedDate

romanbychkoosf commented on Opportunity Summarizer/force-app/main/default/classes/OpportunitySummarizer.cls:
```diff
@@ -0,0 +1,147 @@
+    * 
+    *****************************************************************************************************************/
+    private void checkAmountMap(Opportunity opp, Map<String, Decimal> stageTotalAmounts) {
+        Decimal totalAmount = stageTotalAmounts.get(opp.StageName);
```
you can also use method
if(stageTotalAmounts.containsKey(opp.StageName) {

}

berkebakkalogluosf committed:
overhaul

- made the batchable class accomodate zero opportunities
- moved the test data creation to a seperate class
- created a seperate test class for the batchable class
